release/seccomp.go

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package release
+
+import (
+	"io/ioutil"
+	"sort"
+	"strings"
+)
+
+var (
+	secCompAvailableActionsPath = "/proc/sys/kernel/seccomp/actions_avail"
+)
+
+var SecCompActions []string
+
+func init() {
+	SecCompActions = getSecCompActions()
+}
+
+func MockSecCompActions(actions []string) (restore func()) {
+	old := SecCompActions
+	SecCompActions = actions
+	return func() { SecCompActions = old }
+}
+
+// SecCompActions returns a sorted list of seccomp actions like
+// []string{"allow", "errno", "kill", "log", "trace", "trap"}.
```

Comment:
It is worth pointing out that the following kernel commit changes `kill` to `kill_thread` (in 4.14 and newer):

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=fd76875ca289a3d4722f266fd2d5532a27083903

This means that seccomp policy will be recompiled when booting between a kernel older than 4.14 and a kernel equal to or newer than 4.14. However, I think that's generally a good process to follow when such a low level kernel seccomp change is detected.